contracts/src/Teleporter/upgrades/TeleporterRegistry.sol

```diff
@@ -37,6 +120,93 @@ contract TeleporterRegistry is WarpProtocolRegistry {
         view
         returns (ITeleporterMessenger)
     {
-        return ITeleporterMessenger(_getAddressFromVersion(_latestVersion));
+        return ITeleporterMessenger(getAddressFromVersion(latestVersion));
+    }
+
+    /**
+     * @dev Gets the {ITeleporterMessenger} contract of the given `version`.
+     * Requirements:
+     *
+     * - `version` must be a valid version registered.
+     */
+    function getTeleporterFromVersion(
+        uint256 version
+    ) external view returns (ITeleporterMessenger) {
+        return ITeleporterMessenger(getAddressFromVersion(version));
+    }
+
+    /**
+     * @dev Gets the address of a protocol version.
+     * Requirements:
+     *
+     * - `version` must be a valid version registered.
+     */
+    function getAddressFromVersion(
+        uint256 version
+    ) public view returns (address) {
+        require(version != 0, "TeleporterRegistry: zero version");
+        address protocolAddress = _versionToAddress[version];
+        require(
+            protocolAddress != address(0),
+            "TeleporterRegistry: version not found"
+        );
+        return protocolAddress;
+    }
+
+    /**
+     * @dev Gets the version of the given `protocolAddress`.
+     * Rrequirements:
+     *
+     * - `protocolAddress` must be a valid protocol address registered.
+     */
+    function getVersionFromAddress(
+        address protocolAddress
+    ) public view returns (uint256) {
+        require(
+            protocolAddress != address(0),
+            "TeleporterRegistry: zero protocol address"
+        );
+        uint256 version = _addressToVersion[protocolAddress];
+        require(version != 0, "TeleporterRegistry: protocol address not found");
+        return version;
+    }
+
+    /**
+     * @dev Adds the new protocol version address to the registry.
+     * Updates latest version if the version is greater than the current latest version.
+     *
+     * Emits a {AddProtocolVersion} event when successful.
+     * Emits a {LatestVersionUpdated} event when a new protocol version greater than the current latest version is added.
+     * Note: `protocolAddress` doesn't have to be a contract address, this is primarily
+     * to support the case of registering a new protocol address meant for a security patch
+     * before the contract is deployed, and prevent the vulnerabilitiy from being exposed before the registry update.
```

Comment:
```suggestion
     * Note: `protocolAddress` doesn't have to be a contract address. This allows for registering a new protocol 
     * address before the contract is deployed. This may be the desired order of events to prevent publishing the 
     * patch for a vulnerability in the current version before the update takes effect. 
```

Nit: The current wording seemed a little cumbersome.

Reply:
updated